### DISCUSSION TITLE
How to debug empty unhandled promise rejection?

### DISCUSSION THREAD

**Discussion by christianinsert** [cypress-io/cypress#19304] [Questions and Help]
In a test spec, I keep getting:
```
CypressError: The following error originated from your application code, not from Cypress. It was caused by an unhandled promise rejection.

  >                                                                                                                                                                        

When Cypress detects uncaught errors originating from your application it will automatically fail the current test. 
```
Yes, the error message is empty.
In the corresponding video, it says `(uncaught exception) undefined:  undefined` - which is not very helpful either.

How can I debug such cases?
I have found out so far that it never happens when running in graphical mode. Whereas in head-less mode it fails always in Electron, but only sometimes in Chrome. But still, without a proper error message, that doesn't help much.

**Comment by pompan129** [comment]
One suggestion that helped me was to add 
```
Cypress.on("uncaught:exception", (err, runnable) => {
console.log("uncaught:exception",err);
  return false;
});
```
at the top of a failing test (above describe()).  I was able to see the stack trace then and debug from there.
